src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -0,0 +1,272 @@
+    }
+
+    public List<String> getMatchedLabels() {
+        return matchedLabels;
+    }
+
+    /*
+     * Helper functions
+     */
+
+    private String getMatchedLabelName(Set<String> repoLabels, String keyword) {
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByName(newMatchedLabels, getName(keyword));
+        newMatchedLabels = filterByGroup(newMatchedLabels, getGroup(keyword));
+        return newMatchedLabels.get(0);
+    }
+
+    private boolean hasExactlyOneMatchedLabel(Set<String> repoLabels, String keyword) {
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByName(newMatchedLabels, getName(keyword));
+        newMatchedLabels = filterByGroup(newMatchedLabels, getGroup(keyword));
+        return newMatchedLabels.size() == 1;
+    }
+
+    private void removeConflictingLabels(String name) {
+        if (hasExclusiveGroup(name)) {
+            String group = getGroup(name);
+            // Remove from addedLabels
+            addedLabels = addedLabels.stream()
+                    .filter(label -> {
+                        String labelGroup = getGroup(label);
+                        return !labelGroup.equals(group);
+                    })
+                    .collect(Collectors.toSet());
+
+            // Add to removedLabels all initialLabels that have conflicting group
+            removedLabels.addAll(initialLabels.stream()
+                    .filter(label -> {
+                        String labelGroup = getGroup(label);
+                        return labelGroup.equals(group) && !removedLabels.contains(name);
+                    })
+                    .collect(Collectors.toSet()));
+        }
+    }
+
+    private boolean hasExclusiveGroup(String name) {
+        return TurboLabel.getDelimiter(name).isPresent() && TurboLabel.getDelimiter(name).get().equals(".");
+    }
+
+    private List<String> filterByName(List<String> repoLabels, String labelName) {
+        return repoLabels
+                .stream()
+                .filter(name -> Utility.containsIgnoreCase(getName(name), labelName))
+                .collect(Collectors.toList());
+    }
+
+    private List<String> filterByGroup(List<String> repoLabels, String labelGroup) {
+        if (labelGroup.equals("")) {
+            return repoLabels;
+        } else {
+            return repoLabels
+                    .stream()
+                    .filter(name -> {
+                        if (hasGroup(name)) {
+                            return Utility.containsIgnoreCase(getGroup(name), labelGroup);
+                        } else {
+                            return false;
+                        }
+                    })
+                    .collect(Collectors.toList());
+        }
+    }
+
+    private String getGroup(String name) {
+        if (hasGroup(name)) {
```

Comment:
Same here. The TurboLabel method returns an optional, which helps disambiguate the group for `label` and `.label`.